partiql-cli/src/test/kotlin/org/partiql/cli/pico/InferCommandTest.kt

```diff
@@ -0,0 +1,362 @@
+                )
+            )
+        )
+        // TODO: Need to throw errors on bad references
+        val TABLE_ERROR = TableDescriptor(DEFAULT_TABLE_NAME, attributes = listOf(ColumnMetadata("_1", StaticType.ANY)))
+    }
+
+    // TODO: Currently, SELECT a.a, a FROM <<{'a': 1}>> AS a isn't working, and it's causing issues. To remediate,
+    //  we need to add the input binding tuple to the PROJ operator's schema. This causes SELECT * FROM anything to fail.
+    @Ignore
+    @ParameterizedTest
+    @ArgumentsSource(TestProvider::class)
+    fun test(tc: TestCase): Unit = when (tc) {
+        is SuccessTestCase -> {
+            val session = PlannerSession(
+                tc.query.hashCode().toString(),
+                userId,
+                tc.catalog,
+                tc.catalogPath,
+                catalogMap,
+                Instant.now()
+            )
+            val result = command.infer(tc.query, session)
+            assert(descriptorEquals(tc.expected, result))
+        }
+        is ErrorTestCase -> {
+            val session = PlannerSession(tc.query.hashCode().toString(), userId, tc.catalog, tc.catalogPath, catalogMap, Instant.now())
+            // TODO: Need to pass in an ErrorHandler and assert the error thrown
+            val result = command.infer(tc.query, session)
+            assert(descriptorEquals(TABLE_ERROR, result)) {
+                println("Expected: $TABLE_ERROR")
+                println("Actual: $result")
+            }
+        }
+    }
+
+    sealed class TestCase() {
+        class SuccessTestCase(
+            val name: String,
+            val query: String,
+            val catalog: String? = null,
+            val catalogPath: List<String> = emptyList(),
+            val expected: ValueDescriptor
+        ) : TestCase() {
+            override fun toString(): String = "$name : $query"
+        }
+
+        class ErrorTestCase(
+            val name: String,
+            val query: String,
+            val catalog: String? = null,
+            val catalogPath: List<String> = emptyList()
+        ) : TestCase() {
+            override fun toString(): String = "$name : $query"
+        }
+    }
+
+    class TestProvider : ArgumentsProvider {
+        override fun provideArguments(context: ExtensionContext?): Stream<out Arguments> {
+            return parameters.map { Arguments.of(it) }.stream()
+        }
+        private val parameters = listOf(
+            ErrorTestCase(
+                name = "Test #1",
+                query = "SELECT * FROM pets"
+            ),
+            ErrorTestCase(
+                name = "Test #2",
+                catalog = CATALOG_AWS,
+                query = "SELECT * FROM pets"
+            ),
+            SuccessTestCase(
+                name = "Test #3",
+                catalog = CATALOG_AWS,
+                catalogPath = listOf("ddb"),
+                query = "SELECT * FROM pets",
+                expected = TABLE_AWS_DDB_PETS
+            ),
```

Comment:
This is failing: 
```
expect : TableDescriptor(name=UNSPECIFIED, attributes=[ColumnMetadata(name=id, type=int), ColumnMetadata(name=breed, type=string)])
actual : TableDescriptor(name=UNSPECIFIED, attributes=[ColumnMetadata(name=pets, type=struct(id: int, breed: string)), ColumnMetadata(name=id, type=int), ColumnMetadata(name=breed, type=string)])
```

Reply:
The reason seems to be, by `*` get resolved to table(pet) + attribute(breed, id).

Reply:
Also, experimented with this a bit: 

---------- resolution combination ---------- 

`SELECT ... FROM aws.ddb.pets` compiles. 
`SELECT ... FROM ddb.pets` compiles
`SELECT ... FROM aws.pets` throws : `org.partiql.lang.ast.passes.SemanticException: Semantic Error: at line -1, column -1: Undefined variable 'aws'.` 

---------------------------------------

-----------non existing ----------------

`SELECT ... FROM aws.ddb.foo` throws : `org.partiql.lang.ast.passes.SemanticException: Semantic Error: at line -1, column -1: Undefined variable 'aws'.` 
`SELECT ... FROM aws.foo` throws : `org.partiql.lang.ast.passes.SemanticException: Semantic Error: at line -1, column -1: Undefined variable 'aws'.` 
`SELECT ... FROM aws.ddb.foo` throws : `org.partiql.lang.ast.passes.SemanticException: Semantic Error: at line -1, column -1: Undefined variable 'aws'.` 
`SELECT ... FROM ddb.foo` throws : `org.partiql.lang.ast.passes.SemanticException: Semantic Error: at line -1, column -1: Undefined variable 'ddb'.` 

------------------

---------- arbitrary depth ---------------

`SELECT * FROM aws.ddb.pets.foo` compiles: but the result is a little werid `TableDescriptor(name=UNSPECIFIED, attributes=[ColumnMetadata(name=foo, type=missing), ColumnMetadata(name=_1, type=missing)])` 

`SELECT * FROM ddb.pets.foo`  throws: `org.partiql.lang.ast.passes.SemanticException: Semantic Error: at line 0, column 0: Unhandled exception occurred.`

------------------